### PR TITLE
Fix expired session data cleanup

### PR DESCRIPTION
**Pull request by tharindu-bandara**
### Proposed changes in this pull request
- Fix https://github.com/wso2/product-is/issues/7392.

## Approach
- When deleting expired session data by chunks, each chunk deletion query needs to be committed per chunk. Also, try-with-resources is used to work with the prepared statement to improve the prepared statement closing.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java [hunk 1]
```diff
@@ -445,26 +443,26 @@ private void removeExpiredSessionData(String sqlQuery) {
...
-                statement = connection.prepareStatement(sqlQuery);
-                statement.setLong(1, currentTime);
-
-                int noOfDeletedRecords = statement.executeUpdate();
-                deleteCompleted = noOfDeletedRecords < deleteChunkSize;
-                totalDeletedEntries += noOfDeletedRecords;
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format("Removed %d expired session records.",
-                            noOfDeletedRecords));
+                try (PreparedStatement statement = connection.prepareStatement(sqlQuery)){
```
Comment (madurangasiriwardena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java]: Formatting issue

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java [hunk 2]
```diff
@@ -445,26 +443,26 @@ private void removeExpiredSessionData(String sqlQuery) {
...
-                log.debug(String.format("Deleted total of %d entries ", totalDeletedEntries));
+                log.debug(String.format("Deleted total of %d entries", totalDeletedEntries));
             }
         } catch (SQLException e) {
             IdentityDatabaseUtil.rollbackTransaction(connection);
-            log.error("Error while removing session data from the database for nano time " + currentTime, e);
+            log.error("Error while removing session data from the database for nano time: " + currentTime, e);
         } finally {
-            IdentityDatabaseUtil.closeAllConnections(connection, null, statement);
+            IdentityDatabaseUtil.closeAllConnections(connection, null, null);
```
Comment (madurangasiriwardena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java]: Let's remove the finally block and enclose the connection in a try with resource block.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java (hunk 1 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java]: Fixed in a5f2647666a6f6cd722efacc141af549d9a14496.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java (hunk 2 above)
Comment (tharindu-bandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java]: Fixed in a5f2647666a6f6cd722efacc141af549d9a14496.

**Pull request merged**